### ISSUE TITLE
fix: dependencies not included in generated package.json for new releases

### ISSUE THREAD

**Issue by eamon0989** [s1seven/schema-tools#235]
Previously when building a project, for example, using `nx run generate-pdf:build`, the build code would be output to `dist/packages/generate-pdf` along with the `package.json` file containing a list of the projects dependencies. See release [0.2.7](https://www.npmjs.com/package/@s1seven/schema-tools-generate-pdf/v/0.2.7?activeTab=code) from 7 months ago.

After upgrading to nx 17 this behavior has changed and the dependencies are no longer bundled, and it appears the option to do so has been deprecated. See the empty dependencies in the package.json file of the latest release [0.2.9](https://www.npmjs.com/package/@s1seven/schema-tools-generate-pdf/v/0.2.9?activeTab=code).

More insight on the deprecation of this functionality can be found in a discussion [here](https://github.com/nrwl/nx/issues/18851).

It seems that we will have to replace the use of `@nx/js:tsc` with one of the other options that they provide, including `webpack`, `esbuild`, `vite` or `remix`.
It seems that with `esbuild` there is no option to generate `d.ts` files, although there is a currently open PR to add this functionality to `@nx/esbuild`.

**Comment by getlarge** [comment]
@eamon0989 check the new ESLint rule `@nx/dependency-checks`.
And if you need help on its usage with schema-tools, just let know.

**Comment by eamon0989** [comment]
Thanks! It seems that previously we left the source `package.json` `dependencies` section empty (e.g. `packages/generate-pdf/package.json` and expected the build step to automatically populate the `dependencies` in the build output. From what I understand of the new ESLint rule and some comments in nx issues that I will link below, we now have to manually populate the source `package.json` file, and that will effectively be copied over to the build output, or run `lint --fix` to do it automatically.

See https://github.com/nrwl/nx/issues/18851#issuecomment-1696953346,
https://github.com/nrwl/nx/issues/13880#issuecomment-1669545059,
and https://github.com/nrwl/nx/issues/13880#issuecomment-1701303212

TLDR: I added the new lint rule and ran `lint --fix` which automatically populates the dependencies, and then just requires some minor tweaks.
